magpie/typedefs.py

```diff
@@ -86,3 +87,12 @@
     ConfigItem = Dict[Str, JSON]
     ConfigList = List[ConfigItem]
     ConfigDict = Dict[Str, Union[Str, ConfigItem, ConfigList, JSON]]
+
+    WebhookTemplateParameters = Dict[Str, AnyValue]
+    WebhookConfigSettings = TypedDict("WebhookConfigSettings", {
+        "name": Str, "action": Str, "method": Str, "url": Str, "format": Str, "payload": JSON
+    })
+    WebhookLoadedSettings = TypedDict("WebhookLoadedSettings", {
+        "name": Str, "method": Str, "url": Str, "format": Str, "payload": JSON
+    })
```

Comment:
Unsure here about the difference between WebhookConfigSettings and WebhookLoadedSettings, and the purpose of having 2 definition of settings?

Reply:
The loaded settings don't have `action` compared to config definitions because of how the field was explicitly omitted by `setup_webhooks`, when they are regrouped by corresponding `action` under a list. These definitions detect that difference.

Was there any reason to omit the `action` in the first place other than because of redundant information?
If this is the case, we could leave it and simplify these types.

Reply:
I see why it was done in 2 definitions now. Yes, `action` was removed because of redundant information, since webhooks are reordered by action. 
I guess we have the choice between using these 2 type definitions, or letting the 'action' value in the webhook config setting, and just not using it in the processing later.

I wonder if using those 2 definitions could become confusing, although both solutions work.
What do you prefer?

Reply:
I think I will leave the action in the dict even if redundant just to simplify everything and to align loaded config with whatever was in the original config. 
I was confused about the missing `action` in the loaded settings already until I realized this processing was accomplished.

Reply:
Alright, that's good, I think it will reduce possible confusion here!